### ISSUE TITLE
No module named '_gdal' in python3.8 

### ISSUE THREAD

**Issue by meteoDaniel** [OSGeo/gdal#2957]
## Expected behavior and actual behavior.
I want to use `gdal.Translate` in python3.8 running in a docker container based on `osgeo/gda:ubuntu-full-latest`

This is my Dockerfile:
```
FROM osgeo/gdal:ubuntu-full-latest

RUN apt-get update
RUN apt-get -yqq --no-install-recommends upgrade
RUN apt-get -yqq --fix-missing install python3.8
RUN update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.8 1

RUN rm /usr/bin/python3
RUN ln -s python3.8 /usr/bin/python3

RUN apt-get update && apt -yqq --fix-missing install python3-pip

COPY ./requirements.txt /opt/requirements.txt
RUN pip3 install --upgrade setuptools wheel
RUN pip3 install -r /opt/requirements.txt

RUN apt-get install -yqq --no-install-recommends libglib2.0-0 libsm6 libxext6 libxrender-dev
```
and `requirements.txt`

```
click==6.7
paramiko==2.4.0
requests2==2.16.0
graypy==1.1.2
opencv-python==4.2.0.34
imutils==0.5.3
SQLAlchemy==1.3.2
psycopg2-binary==2.8.4
geoalchemy2==0.6.3
pyproj==2.6.1.post1
faster-than-requests==0.9.6
```

And actually I am receiving this error when importing `gdal`: 

```
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/osgeo/__init__.py", line 18, in swig_import_helper
    fp, pathname, description = imp.find_module('_gdal', [dirname(__file__)])
  File "/usr/lib/python3.8/imp.py", line 296, in find_module
    raise ImportError(_ERR_MSG.format(name), name=name)
ImportError: No module named '_gdal'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "src/pipelines/parse_lightning_strike_data_germany.py", line 4, in <module>
    from src.models.parse_lightning_data import \
  File "/app/src/models/parse_lightning_data.py", line 5, in <module>
    import gdal
  File "/usr/lib/python3/dist-packages/gdal.py", line 2, in <module>
    from osgeo.gdal import deprecation_warn
  File "/usr/lib/python3/dist-packages/osgeo/__init__.py", line 41, in <module>
    _gdal = swig_import_helper()
  File "/usr/lib/python3/dist-packages/osgeo/__init__.py", line 20, in swig_import_helper
    import _gdal
ModuleNotFoundError: No module named '_gdal'
```
I was not able to find the answer via google, stackoverflow and this github repo. 

I hope anyone will know how to fix it.

**Comment by rouault** [comment]
I tried to reproduce your steps one by one and issue python3 -c "import gdal" and it works fine, with the latest osgeo/gdal:ubuntu-full-latest based on Ubuntu 20.04, which already uses python3.8, so your dance with installing python3.8 is not needed. If you use an older image based on ubuntu 18.04 that might explain it, since the /usr/lib/python3/dist-packages/osgeo/ would be built against  a different python version.

Note however that you may experience crashes at runtime, when mixing pyproj and the GDAL python bindings, since pyproj comes with a bundled libproj library, that might conflict with the one of the osgeo/gdal:ubuntu-full-latest that is built from PROJ master, but this is a different issue than the one you get. In which case you might want to build pyproj from source against the libproj of the osgeo/gdal:ubuntu-full-latest  image

**Comment by ibaris** [comment]
I experience the same issue with just
```shell
conda create -n gdaltest -c conda-forge gdal
```

**Comment by jratike80** [comment]
See what has worked couple of months ago https://gis.stackexchange.com/questions/461363/modulenotfounderror-no-module-named-gdal-in-ubuntu-20-04-and-python-3-8/461391#461391

**Comment by ibaris** [comment]
> See what has worked couple of months ago https://gis.stackexchange.com/questions/461363/modulenotfounderror-no-module-named-gdal-in-ubuntu-20-04-and-python-3-8/461391#461391

Thank you very much. I already tried this solution but it does not work for me.

**Comment by jratike80** [comment]
I tried also myself and even I know that those commands have worked for me on this same computer, now I do get errors. The traceback below:

<details><summary>Details</summary>
<p>

from osgeo import gdal
Traceback (most recent call last):
  File "C:\Users\JRAHKONEN\Miniconda3\envs\gdaltest\lib\site-packages\osgeo\__init__.py", line 30, in swig_import_helper
    return importlib.import_module(mname)
  File "C:\Users\JRAHKONEN\Miniconda3\envs\gdaltest\lib\importlib\__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 657, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 556, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 1166, in create_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
ImportError: DLL load failed while importing _gdal: Määritettyä proseduuria ei löydy.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\JRAHKONEN\Miniconda3\envs\gdaltest\lib\site-packages\osgeo\__init__.py", line 46, in <module>
    _gdal = swig_import_helper()
  File "C:\Users\JRAHKONEN\Miniconda3\envs\gdaltest\lib\site-packages\osgeo\__init__.py", line 43, in swig_import_helper
    return importlib.import_module('_gdal')
  File "C:\Users\JRAHKONEN\Miniconda3\envs\gdaltest\lib\importlib\__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ModuleNotFoundError: No module named '_gdal'

</p>
</details>

**Comment by ibaris** [comment]
> I tried also myself and even I know that those commands have worked for me on this same computer, now I do get errors. The traceback below:
> 
> Details

I got the same error.

**Comment by rouault** [comment]
@jratike80  Do GDAL native command line utilities work (gdalinfo, etc.) ?  Which python3 version is used? (python3 --version)  What is the output of ``conda list``

**Comment by jratike80** [comment]
No, the GDAL native tools do not work in this environment. Windows throws an error about missing "gssapi64.dll". However, I found an old environment that I have used for testing something, and there the native tools do run, but importing gdal into Python still fails. In that half-working environment the Python version is 3.10.2. The errors which are reported from the `File <frozen importlib._bootstrap>` are the same than in the traceback that I attached to my previous comment, but the line numbers differ.

In the fresh environment that I created with the commands that I took from the gis.stackexchange answer the Python version is 3.8.17. 

<details><summary>conda list</summary>
<p>


 packages in environment at C:\Users\JRAHKONEN\Miniconda3\envs\gdaltest:

 Name                    Version                   Build  Channel
blas                      1.0                         mkl
blosc                     1.21.3               h6c2663c_0
boost-cpp                 1.73.0              h2bbff1b_12
bzip2                     1.0.8                he774522_0
ca-certificates           2023.05.30           haa95532_0
cairo                     1.16.0               haedb8bc_5
cfitsio                   3.470                h2bbff1b_7
expat                     2.4.9                h6c2663c_0
fontconfig                2.14.1               h9c4af85_2
freetype                  2.12.1               ha860e81_0
freexl                    1.0.6                h2bbff1b_0
gdal                      3.6.2            py38h7670e6c_3
geos                      3.8.0                h33f27b4_0
geotiff                   1.7.0                h4545760_1
glib                      2.69.1               h5dc1a3c_2
hdf4                      4.2.13               h712560f_2
hdf5                      1.12.1               h51c971a_3
icc_rt                    2022.1.0             h6049295_2
intel-openmp              2023.1.0         h59b6b97_46319
jpeg                      9e                   h2bbff1b_1
kealib                    1.5.0                hde4a422_1
lcms2                     2.12                 h83e58a3_0
lerc                      3.0                  hd77b12b_0
libboost                  1.73.0              h6c2663c_12
libcurl                   8.1.1                h86230a5_0
libdeflate                1.17                 h2bbff1b_0
libffi                    3.4.4                hd77b12b_0
libgdal                   3.6.2                h676b171_3
libiconv                  1.16                 h2bbff1b_2
libkml                    1.3.0                h63940dd_6
libnetcdf                 4.8.1                h6685c40_4
libpng                    1.6.39               h8cc25b3_0
libpq                     12.15                h906ac69_0
libspatialite             4.3.0a              h6ec8781_23
libssh2                   1.10.0               he2ea4bf_2
libtiff                   4.5.0                h6c2663c_2
libwebp-base              1.2.4                h2bbff1b_1
libxml2                   2.10.3               h0ad7f3c_0
libzip                    1.8.0                h289538f_1
lz4-c                     1.9.4                h2bbff1b_0
mkl                       2023.1.0         h8bd8f75_46356
mkl-service               2.4.0            py38h2bbff1b_1
mkl_fft                   1.3.6            py38hf11a4ad_1
mkl_random                1.2.2            py38hf11a4ad_1
numpy                     1.24.3           py38h79a8e48_1
numpy-base                1.24.3           py38h8a87ada_1
openjpeg                  2.4.0                h4fc8c34_0
openssl                   3.0.10               h2bbff1b_0
pcre                      8.45                 hd77b12b_0
pcre2                     10.42                h0ff8eda_0
pip                       23.2.1           py38haa95532_0
pixman                    0.40.0               h2bbff1b_1
poppler                   22.12.0              h268424c_0
poppler-data              0.4.11               haa95532_1
proj                      6.2.1                h3758d61_0
python                    3.8.17               h1aa4202_0
qhull                     2020.2               h59b6b97_2
setuptools                68.0.0           py38haa95532_0
sqlite                    3.41.2               h2bbff1b_0
tbb                       2021.8.0             h59b6b97_0
tiledb                    2.3.3                hd8964de_3
vc                        14.2                 h21ff451_1
vs2015_runtime            14.27.29016          h5e58377_2
wheel                     0.38.4           py38haa95532_0
xerces-c                  3.2.4                hd77b12b_0
xz                        5.4.2                h8cc25b3_0
zlib                      1.2.13               h8cc25b3_0
zstd                      1.5.5                hd43e919_0

</p>
</details> 

Also the following message appeared when I tried the import for the first time. It was silenced by setting the env, but it did not make the import to succeed.

> On Windows, with Python >= 3.8, DLLs are no longer imported from the PATH.
> If gdalXXX.dll is in the PATH, then set the USE_PATH_FOR_GDAL_PYTHON=YES environment variable
> to feed the PATH into os.add_dll_directory().
>
